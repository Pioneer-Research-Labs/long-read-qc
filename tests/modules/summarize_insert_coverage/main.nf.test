nextflow_process {

    name "Test Process summarize_insert_coverage"
    script "modules/summarize_insert_coverage/main.nf"
    process "summarize_insert_coverage"

    test("Should generate insert coverage summaries and plots") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file('${projectDir}/test_data/insert_coverage.tsv')
                """
            }
        }

        then {
            assert process.success
            assert process.out.size() == 2
            assert process.out[0][0].size() == 3
            assert process.out[1][0].size() == 2
        }

    }

}
