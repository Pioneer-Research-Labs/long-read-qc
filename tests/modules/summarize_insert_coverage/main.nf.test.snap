{
    "Should generate insert coverage summaries and plots": {
        "content": [
            "concatenated_insert_coverage.csv:md5,b62405adff9309bcadda85a4fa767fe2",
            "full_genes_per_fragment.csv:md5,b62405adff9309bcadda85a4fa767fe2",
            "full_genes_per_fragment.png:md5,68817eb7f9235aca56b9c48198c36339",
            "partial_genes_per_fragment.csv:md5,b62405adff9309bcadda85a4fa767fe2",
            "partial_genes_per_fragment.png:md5,424ae91166dfb94e0f5deccb241ab76e"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T13:43:48.538585"
    }
}